fenix/app/src/test/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckTelemetryMiddlewareTest.kt

```diff
@@ -0,0 +1,141 @@
+
+import io.mockk.mockk
+import mozilla.components.service.glean.testing.GleanTestRule
+import mozilla.components.support.test.ext.joinBlocking
+import mozilla.components.support.test.libstate.ext.waitUntilIdle
+import mozilla.components.support.test.robolectric.testContext
+import mozilla.components.support.test.rule.MainCoroutineRule
+import org.junit.Assert.assertNotNull
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.GleanMetrics.Shopping
+import org.mozilla.fenix.helpers.FenixRobolectricTestRunner
+import org.mozilla.fenix.shopping.store.ReviewQualityCheckAction
+import org.mozilla.fenix.shopping.store.ReviewQualityCheckState
+import org.mozilla.fenix.shopping.store.ReviewQualityCheckStore
+
+@RunWith(FenixRobolectricTestRunner::class)
+class ReviewQualityCheckTelemetryMiddlewareTest {
+
+    @get:Rule
+    val gleanTestRule = GleanTestRule(testContext)
+
+    @get:Rule
+    val coroutinesTestRule = MainCoroutineRule()
+    private val dispatcher = coroutinesTestRule.testDispatcher
+    private val scope = coroutinesTestRule.scope
+    private lateinit var store: ReviewQualityCheckStore
+
+    @Before
+    fun setup() {
+        store = ReviewQualityCheckStore(
+            middleware = listOf(
+                ReviewQualityCheckTelemetryMiddleware(
+                    scope = scope,
+                ),
+            ),
+        )
+        store.waitUntilIdle()
+        dispatcher.scheduler.advanceUntilIdle()
+    }
+
+    @Test
+    fun `WHEN the user opts in the feature THEN the opt in event is recorded`() {
+        store.dispatch(ReviewQualityCheckAction.OptIn).joinBlocking()
+        store.waitUntilIdle()
+
+        assertNotNull(Shopping.bottomSheetOptInAccepted.testGetValue())
+    }
+
+    @Test
+    fun `WHEN the contextual onboarding is displayed THEN the onboarding event is recorded`() {
+        store.dispatch(ReviewQualityCheckAction.OnboardingDisplayed).joinBlocking()
+        store.waitUntilIdle()
+
+        assertNotNull(Shopping.bottomSheetOnboardingDisplayed.testGetValue())
+    }
+
+    @Test
+    fun `WHEN the bottom sheet is closed THEN the bottom sheet closed event is recorded`() {
+        store.dispatch(ReviewQualityCheckAction.BottomSheetClosed).joinBlocking()
+        store.waitUntilIdle()
+
+        assertNotNull(Shopping.bottomSheetClosed.testGetValue())
+    }
+
+    @Test
+    fun `WHEN the bottom sheet is displayed THEN the bottom sheet displayed event is recorded`() {
+        store.dispatch(ReviewQualityCheckAction.BottomSheetDisplayed).joinBlocking()
+        store.waitUntilIdle()
+
+        assertNotNull(Shopping.bottomSheetDisplayed.testGetValue())
+    }
+
+    @Test
+    fun `WHEN the learn more link from the explainer card is clicked THEN the explainer learn more event is recorded`() {
+        val openedLink = mockk<ReviewQualityCheckState.LinkType.ExplainerLearnMore>(relaxed = true)
```

Comment:
Let's not mockk this, we can create an object with a random url here, removing the need to use mockk in this test class altogether.